assets/openapi.yaml

```diff
@@ -152,6 +180,144 @@ components:
+      properties:
+        id:
+          type: string
+          format: uuid
+          example: 20210510100104650::88b5f678-fef5-447b-af15-f21e367e6d8c
+        account_id:
+          type: string
+          format: uuid
+          example: c8f1ef5d-edc0-4f23-9ee4-378f19cb92a4
+        activity_type:
+          $ref: '#/components/schemas/ActivityType'
+    TradeActivity:
+      title: TradeActivity
+      allOf:
+        - $ref: '#/components/schemas/Activity'
+        - type: object
+          properties:
+            transaction_time:
+              type: string
+              format: timestamp
+              example: '2021-05-10T14:01:04.650275Z'
+            type:
+              type: string
+              enum:
+                - fill
+                - partial_fill
+              example: fill
+            price:
+              type: string
+              example: 3.1415
+            qty:
+              type: string
+              example: 0.38921
+            side:
+              type: string
+              enum:
+                - buy
+                - sell
+              example: sell
+            symbol:
+              type: string
+              example: AAPL
+            leaves_qty:
+              type: string
+              example: 0.5123
+            order_id:
+              type: string
+              format: uuid
+              example: fe060a1b-5b45-4eba-ba46-c3a3345d8255
+            cum_qty:
+              type: string
+              example: 0.9723
+            order_status:
+              type: string
+              enum:
+                - new
+                - partially_filled
+                - filled
+                - done_for_day
+                - canceled
+                - expired
+                - replaced
+                - pending_cancel
+                - pending_replace
+                - accepted
+                - pending_new
+                - accepted_for_bidding
+                - stopped
+                - rejected
+                - suspended
+                - calculated
+              example: filled
+    NonTradeActivity:
+      title: NonTradeActivity
+      allOf:
+        - $ref: '#/components/schemas/Activity'
+        - type: object
+          properties:
+            date:
+              type: string
+              format: date
+              example: '2021-05-21'
+            net_amount:
+              type: string
+              example: 1234
+            description:
+              type: string
+              example: Example description
+            status:
+              type: string
+              enum:
+                - executed
+                - correct
+                - canceled
+              example: executed
+            symbol:
+              type: string
+              example: AAPL
+            qty:
+              type: string
+              example: 0.38921
+            per_share_amount:
+              type: string
+              example: 0.38921
```

Comment:
arent these numbers?

Reply:
from our api it comes back as a string containing a number

Reply:
oh right.

> 
	// Qty and PerShareAmount not defined as decimals because decimal values cannot
	// be excluded with omitempty. https://github.com/shopspring/decimal/issues/41